keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -254,33 +258,29 @@ Metadata:
   - cosi.io/finalizer [3]
 spec:
   bucketClassName: [4]
-  supportedProtocols: [5]
-  releasePolicy: {"Delete", "Retain"} [6]
-  bucketRef: [7]
+	bucketRef: [5]
     name:
     namespace:
-  secretName: [8]
-  accessMode: {"rw", "ro"} [9]
+  secretRef: [6]
+  objectBucketName: [7]
 status:
-  bucketAttributes: <map[string]string> [10]
-  phase: {"Bound", "Released", "Failed", "Errored"} [11]
+  bucketAttributes: <map[string]string> [8]
+  phase: [9]
   conditions:
 ```
-1. `name`: Generated in the pattern of `“bucket-”<BUCKET-NAMESPACE>"-"<BUCKET-NAME>`. We may validate the length of the `Bucket` name and namespace to ensure that this _metadata.name_ fits.
-1. `labels`: central controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
+1. `name`: Generated in the pattern of `<BUCKET-CLASS-NAME>'-'<RANDOM-SUFFIX>`. 
+1. `labels`: COSI controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
 1. `bucketClassName`: Name of the associated `BucketClass`.
-1. `supportedProtocols`:  String array of protocols (e.g. s3, gcs, swift, etc.) supported by the associated object store.
-1. `releasePolicy`: the release policy defined in the associated BucketClass (see [BucketClass](#BucketClass) for more information).
 1. `bucketRef`: the name & namespace of the associated `Bucket`. For both brownfield cases, `bucketRef` is set by an admin and names the bucket to be accessed. For greenfield, it is added by the COSI controller.
-1. `secretName`: the name of the app secret in the `Bucket`'s namespace, as define in `bucketRef.namespace`. For static brownfield, `secretName` is set by an admin; otherwise it is added by the COSI controller.
-1. `accessMode`: The level of access granted to the credentials stored in `secretRef`, one of "read only" or "read/write".
+1. `secretRef`: the name and namespace of the source secret in the `Bucket`'s namespace, as define in `bucketRef.namespace`.  This `secret` may be generated by the driver or written by an admin.
```

Comment:
Maybe some rewording:
"`secretRef`: the name and namespace of the secret to be copied to the `Bucket`'s namespace.  This `secret` may be generated by the driver or created by an admin."